authLdap.php

```diff
@@ -263,39 +262,40 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
         }
 
         $uid = authLdap_get_uid($realuid);
-        $role = '';
-
-        // we only need this if either LDAP groups are disabled or
-        // if the WordPress role of the user overrides LDAP groups
-        if (!$authLDAPGroupEnable || !$authLDAPGroupOverUser) {
-            $role = authLdap_user_role($uid);
-        }
+        $roles = authLdap_user_roles($uid);
 
         // do LDAP group mapping if needed
-        // (if LDAP groups override worpress user role, $role is still empty)
-        if (empty($role) && $authLDAPGroupEnable) {
-            $role = authLdap_groupmap($realuid, $dn);
-            authLdap_debug('role from group mapping: ' . $role);
+        if ($authLDAPGroupEnable) {
+            $roles_ldap = authLdap_groupmap($realuid, $dn);
+            authLdap_debug('roles from group mapping: ' . implode(', ', $roles_ldap));
+
+            if ($authLDAPGroupOverUser) {
+                $roles = $roles_ldap;
+            } else {
+                $roles = array_unique(array_merge($roles, $roles_ldap));
+            }
```

Comment:
This if-condition is obsolete, as theroles are assigned above. And merging wordpress-roles with the LDAP-roles is critical, as we *either* use LDAP-roles *or* Wordpress roles. Otherwise it will get messy for the administration